packages/app/src/components/inspector/PickerPatterns.tsx

```diff
@@ -2,15 +2,29 @@ import type Sketch from '@sketch-hq/sketch-file-format-ts';
 import { Divider, getPatternBackground, Select } from 'noya-designsystem';
 import { FileMap } from 'noya-sketch-file';
 import { memo } from 'react';
+import { useObjectURL } from '../../hooks/useObjectURL';
 import * as InspectorPrimitives from '../inspector/InspectorPrimitives';
 import { GridSmall, Square } from './PickerAssetGrid';
 
 interface Props {
   fileImages: FileMap;
   imageAssets: (Sketch.FileRef | Sketch.DataRef)[];
-  onChange: (color: Sketch.FileRef | Sketch.DataRef) => void;
+  onChange: (file: Sketch.FileRef | Sketch.DataRef) => void;
 }
 
+const PatternSquare = memo(
+  ({
+    image,
+    onClick,
+  }: {
+    image: ArrayBuffer | string;
```

Comment:
I think we do want to pass `item` in here after all, so that `onClick` doesn't need to be created inline below. E.g.

```tsx
<Square onClick={useCallback(() => onClick(item)} />
```

This will currently not get memoized, since we redefine `onClick` every render of `<PatternSquare onClick={() => onChange(item)} />`